### PR TITLE
[Core] Minor correction on Parameters settings on solving strategies

### PR DESCRIPTION
**Pull request by loumalouomega**
This adds some minor corrections on settings on solving strategies. This is necessary for making properly work #3185 @KratosMultiphysics/implementation-committee 
 
(There is a way to add as reviewer the @KratosMultiphysics/implementation-committee ?)

### CODE REVIEW

File: kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h [hunk 1]
```diff
@@ -73,7 +73,8 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
         : mpResponseFunction(pResponseFunction)
     {
         Parameters default_parameters(R"({
-            "scheme_type": "bossak",
+            "scheme_type" : "ResidualBasedAdjointBossakScheme",
+            "scheme_type" : "bossak",
```
Comment (jcotela) [inline kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h]: duplicate key "scheme_type" here (this is not valid json)

**Comment by jcotela** [comment]
@loumalouomega I have adjusted the permissions for the @KratosMultiphysics/implementation-committee. You should be able to request it reviews now.

**Comment by loumalouomega** [comment]
Many thanks

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_slip.h [hunk 2]
```diff
@@ -146,10 +146,11 @@ class ResidualBasedEliminationBuilderAndSolverSlip
         // Validate default parameters
         Parameters default_parameters = Parameters(R"(
         {
-            "domain_size" : 3,
-            "variable_x" : "VELOCITY_X",
-            "variable_y" : "VELOCITY_Y",
-            "variable_z" : "VELOCITY_Z"
+            "builder_and_solver_type" : "ResidualBasedEliminationBuilderAndSolverSlip",
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_slip.h]: why don't you use `name` or `type` for this?

Should simplify the handling in the factories and also avoid name-clases

Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_slip.h]: Well, because when constructing a whole strategy you need to differentiate them

Comment (pooyan-dadvand) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_slip.h]: In favor of @philbucher comment

```json
builder_and_solver:{
    "name": "...."
}

Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_slip.h]: Ok, but this will require some extra work

**Comment by loumalouomega** [comment]
Done @philbucher

**Comment by loumalouomega** [comment]
Ping

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h [hunk 3]
```diff
@@ -89,11 +89,13 @@ class ResidualBasedPseudoStaticDisplacementScheme
      * @param ThisParameters Parameters with the Rayleigh variable
      */
     explicit ResidualBasedPseudoStaticDisplacementScheme(Parameters ThisParameters)
-      :DerivedBaseType(0.0)
+      :DerivedBaseType(0.0),
+       mRayleighBeta(NODAL_MAUX)
```
Comment (philbucher) [inline kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: what does this have to do with this PR?

File: kratos/solving_strategies/convergencecriterias/displacement_criteria.h [hunk 4]
```diff
@@ -82,15 +83,38 @@ class DisplacementCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpa
...
     ///@{
 
+    /** Constructor.
+    */
+    explicit DisplacementCriteria(Kratos::Parameters Settings)
+        : BaseType()
+    {
+        if (Settings.Has("displacement_absolute_tolerance")) {
+            mAlwaysConvergedNorm = Settings["displacement_absolute_tolerance"].GetDouble();
+        } else if (Settings.Has("absolute_tolerance")) {
```
Comment (philbucher) [inline kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: Why either or?

**Review by philbucher — COMMENTED** [review]
why do you add new constructors to the conv-criteria?
I am not sure if it is a good idea that the settings can come from different keys in the json
=> this makes it confusing for users

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: Well, this PR is to make easier to review #3185

File: kratos/solving_strategies/convergencecriterias/displacement_criteria.h (hunk 4 above)
Comment (loumalouomega) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: What_?

Comment (philbucher) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: why do you get `mAlwaysConvergedNorm` either from `displacement_absolute_tolerance` OR from `absolute_tolerance`?
I don't think this is a good ides?

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h (hunk 3 above)
Comment (philbucher) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: I checked the other PR but I don't understand why this changes

Comment (loumalouomega) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: BEcause the RAYLEIGHT_BETA is not defined in the core, and it gives an error when running cpp tests

File: kratos/solving_strategies/convergencecriterias/displacement_criteria.h (hunk 4 above)
Comment (loumalouomega) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: The names are so confusing that I not sure about anything right now

Comment (philbucher) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: Then I think you should stick to our usual convention

anyway this is not the scope of this PR

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h (hunk 3 above)
Comment (philbucher) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: why did you add it then in the first place?

I am confused ...

File: kratos/solving_strategies/convergencecriterias/displacement_criteria.h (hunk 4 above)
Comment (loumalouomega) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: I am quite sure that it follows our convention

File: kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h:93]: To be able to create tests in #3185

File: kratos/solving_strategies/convergencecriterias/displacement_criteria.h (hunk 4 above)
Comment (philbucher) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: the double-definition does not follow our convention => this is the thing I think we should avoid

So again my question, why doing this? :)
I guess in order to be independent of which criteria you create?

Comment (loumalouomega) [reply kratos/solving_strategies/convergencecriterias/displacement_criteria.h:93]: Well, a priori you don't know the criterias you are combining

File: kratos/solving_strategies/strategies/solving_strategy.h [hunk 5]
```diff
@@ -109,16 +109,27 @@ class SolvingStrategy
...
-    /** Constructor.
+    /**
+     * @brief Default constructor. (with parameters)
+     * @param rModelPart The model part of the problem
+     * @param ThisParameters The configuration parameters
+     */
+    explicit SolvingStrategy(ModelPart& rModelPart, Parameters ThisParameters)
+        : mrModelPart(rModelPart)
+    {
+        const bool move_mesh_flag = ThisParameters.Has("move_mesh_flag") ? ThisParameters["move_mesh_flag"].GetBool() : false;
```
Comment (RiccardoRossi) [inline kratos/solving_strategies/strategies/solving_strategy.h:120]: missing validate

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/solving_strategy.h:120]: There can not be validate , look #3185, the base class constructor is called in the derived class, so it can not validate or it will crash

Comment (loumalouomega) [reply kratos/solving_strategies/strategies/solving_strategy.h:120]: That's why HAs is used instead

**Comment by loumalouomega** [comment]
Ping

**Comment by philbucher** [comment]
to me looks ok, will approve after the release branch is made

**Comment by loumalouomega** [comment]
Ping @philbucher

**Comment by loumalouomega** [comment]
Ping @philbucher

**Review by philbucher — APPROVED** [review]
sorry didn't see the ping

since none complained I will approve this now, seems like everyone raised their concerns

**Comment by philbucher** [comment]
Thanks for the efforts and patience :)

**Comment by philbucher** [comment]
this makes some tests in fluid fail

**Comment by loumalouomega** [comment]
Ups, I will do a Hotfix

**Comment by philbucher** [comment]
I think its the tests that use the hdf5-app
